### DISCUSSION TITLE
Aligning dictionary items without using a grid

### DISCUSSION THREAD

**Discussion by JCGoran** [typst/typst#2332] [Q&A]
So I have some text that is basically in the form of a typst dictionary (key-value pairs), and I would like to display this on a single line, with all of the keys to be aligned one way, and all of the values being aligned another way, with a single (visual) separator in between (custom whitespace).
I know I can do this be partially achieved using the `grid` function, but there are two issues that arise with a `grid`:

1. the text selection first selects one column (the keys), and then the other one (the values); I would prefer it if it was selectable as `key1 [WHITESPACE] value1` and so on.
2. if any of the values overflow into the next line, the layout looks broken, kind of like this:
```plaintext
key1 this is a value for
key2 key1
key3 value2
...
```
rather than inserting some whitespace to match the key:
```plaintext
key1 this is a value for
     key1
key2 value2
key3 value3
...
```

On the other hand, if I try putting each element in its own grid, the keys and the values will not align according to a single vertical line, which sort of ruins the look, so I'm currently using the "put all keys in one `grid` column, and all values in another, mess with text and page properties until no newlines occur", but I'm curious if there's a way to achieve what I described above.

**Comment by JCGoran** [comment]
After reading the docs more thoroughly, it seems one way to do this is to:

1. put all of the contents of the dictionary into an array:
```typst
#let arr = ()
#for (key, value) in d.pairs() {
  arr.push(key)
  arr.push(value)
}
```
2. use a `table`  instead of a grid:
```typst
table(
    columns: 2, 
    align: (right, left),
    stroke: none,
   ..arr
)
```
3. tweak the `column-gutter` and `row-gutter` parameters of the `table` until the output is satisfactory

**Reply by lphoogenboom** [reply]
Hi, what is the use of the `..` before `arr` in the table?

**Reply by rwmpelstilzchen** [reply]
https://typst.app/docs/reference/foundations/arguments/#argument-sinks
